presto-hive/src/main/java/com/facebook/presto/hive/HivePageSource.java

```diff
@@ -111,6 +112,7 @@
 
     private final ConnectorPageSource delegate;
     private boolean filterAndProjectPushedDown;
+    private final boolean needsColumnMapping;
```

Comment:
Is there any reason why the functionality is bolted into this class?  I think it would be much simpler to read and understand if we split the version where we `filterAndProjectPushedDown` into a separate class.  What tells me we can do this is the if check on `filterAndProjectPushedDown`, which is a feature toggle that short circuits the entire `getNextPage` method.